plugins/catalog-backend-module-time-saver-processor/src/processor/TimeSaverProcessor.ts

```diff
@@ -0,0 +1,59 @@
+import { Entity } from '@backstage/catalog-model';
+import { CatalogProcessor } from '@backstage/plugin-catalog-node';
+import { JsonValue } from '@backstage/types';
+
+type Substitute = {
+  engineering: Record<string, number>;
+};
+
+function isValidSubstitute(
+  substitute: JsonValue | undefined,
+): substitute is Substitute {
+  return (
+    typeof substitute === 'object' &&
+    !Array.isArray(substitute) &&
+    substitute !== null &&
+    substitute !== undefined &&
+    substitute.engineering !== undefined
+  );
+}
+
+export class TimeSaverProcessor implements CatalogProcessor {
```

Comment:
SUGGESTION: we may add optional logger and include debug log informing that processor is triggered

Reply:
No problem, I'll add this now.